llvm/lib/Target/DirectX/DXILMetadata.cpp

```diff
@@ -40,6 +40,15 @@ void ValidatorVersionMD::update(VersionTuple ValidatorVer) {
 
 bool ValidatorVersionMD::isEmpty() { return Entry->getNumOperands() == 0; }
 
+VersionTuple ValidatorVersionMD::getAsVersionTuple() {
+  if (isEmpty())
+    return VersionTuple(1, 0);
+  auto *ValVerMD = cast<MDNode>(Entry->getOperand(0));
```

Comment:
Just to be sure, will Entry be defined for library shaders where there may not necessarily be an entry point?

Reply:
Yes. Entry is always intialized with getOrInsertNamedMetadata.